presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -5,35 +5,548 @@ Tuning Presto
+
+
+``prefer_streaming_operators``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Prefer source table layouts that produce streaming operators. Setting this property will allow workers not to wait for chunks of data to start processing them while scanning tables. This may cause faster processing  with lower latency and downtime but some operators may do things more efficiently when working with chunks of data.
+
+
+``push_table_write_through_union``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``optimizer.push-table-write-through-union`` (``true``)
+ * **Description:** See :ref:`optimizer.push-table-writethrough-union <tuning-pref-optimizer>`.
+
+
+``query_max_memory``
+^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (data size)
+ * **Default value:** ``query.max-memory`` (``20 GB``)
+ * **Description:** This property can be use to be nice to the cluster for example when our query is not as important then the usual cluster routines. Setting this value to smaller then server property ``query.max-memory`` will cause server to drop session query if it will require more then ``query_max_memory`` memory instead of ``query.max-memory``. On the other hand setting this value to higher then ``query.max-memory`` will not have effect at all.
+
+
+``query_max_run_time``
+^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``String`` (duration)
+ * **Default value:** ``query.max-run-time`` (``100 d``)
+ * **Description:** The default value of this is defined by server. If expected query processing time is higher then property ``query.max-run-time`` it's crucial to set this session property - otherwise there is a risk of dropping all result of long processing after ``query.max-run-time`` ends. Session may also set this value to lower than ``query.max-run-time`` in order to crosscheck for bugs in queries. In may be particularly use full when setting up session with very large number of queries each of which should take very short time in order to be able to end all of queries in acceptable time. Even in this scenario it's crucial though, to set this value to much higher value than average query time to avoid problems with outliers (some queries may randomly take much longer then other due to cluster load and many other circumstances).
+
+
+``resource_overcommit``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+ * **Type:** ``Boolean``
+ * **Default value:** ``false``
+ * **Description:** Use resources which are not guaranteed to be available to the query. By setting this property you allow to exceed limits of memory available per query processing and session. This may cause resources to be used more efficiently allowing to  but may cause some indeterministic query drops due to lacking memory on machine. perform more demanding queries
```

Comment:
Use resources that are not guaranteed to be available to a query.  This property allows you to exceed the limits of memory available per query and session.  It may allow resources to be used more efficiency, bu may also cause non-deterministic query drops  due to insufficient memory on machine.  It can be particularly useful for performing more demanding queries.